agent/agenttest/agent.go

```diff
@@ -0,0 +1,97 @@
+package agenttest
+
+import (
+	"context"
+	"net/url"
+	"sync"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+	"github.com/coder/coder/v2/agent"
+	"github.com/coder/coder/v2/coderd/coderdtest"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/agentsdk"
+)
+
+// Agent is a small wrapper around an agent for use in tests.
+type Agent struct {
+	agent       agent.Agent
+	agentClient *agentsdk.Client
+	resources   []codersdk.WorkspaceResource
+	waiter      func(*codersdk.Client, uuid.UUID, ...string) []codersdk.WorkspaceResource
```

Comment:
why is this a field rather than just a normal method?

Reply:
No longer exists, use `coderdtest.AwaitWorkspaceAgents` instead.